CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift

```diff
@@ -0,0 +1,233 @@
+    @IBOutlet weak var thirdDigit: UILabel!
+    @IBOutlet weak var fourthDigit: UILabel!
+    @IBOutlet weak var fifthDigit: UILabel!
+    @IBOutlet weak var sixthDigit: UILabel!
+    
+    @IBOutlet weak var firstSeparator: UIView!
+    @IBOutlet weak var secondSeparator: UIView!
+    @IBOutlet weak var thirdSeparator: UIView!
+    @IBOutlet weak var fourthSeparator: UIView!
+    @IBOutlet weak var fifthSeparator: UIView!
+    
+    @IBOutlet weak var borderView: UIView!
+    @IBOutlet weak var countDownDecription: UILabel!
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    
+    private var digits: [UILabel]?
+    private var counter: Double = 0
+    private var timer: Timer?
+    private var colorInput: UIColor?
+    
+    var viewModel: ViewPinViewModel?
+    
+    // MARK: - Init
+    /// Initialization of NIViewPinView
+    /// To be used when creating the PIN view programatically
+    /// - Parameters:
+    ///   - input: input needed for the PIN visualization
+    ///   - timer: seconds needed for the PIN visualization
+    public init(input: NIInput, timer: Double, color: UIColor? = nil, completion: @escaping (NISuccessResponse?, NIErrorResponse?, @escaping () -> Void) -> Void) {
+        counter = timer
+        colorInput = color
+        viewModel = ViewPinViewModel(input: input)
+        viewModel?.callback = completion
+        super.init(frame: .zero)
+        fromNib()
+        hideUI(true)
+        GlobalConfig.shared.language = input.displayAttributes?.language
+        
+        if self.borderView != nil {
+            activityIndicator.startAnimating()
+            updateUI()
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        fromNib()
+    }
+    
+    // MARK: - Public
+    /// Set the input for the NIViewPinView
+    /// To be used ONLY if NIViewPinView is added in storyboard or xib
+    /// - Parameters:
+    ///   - input: input needed for the PIN visualization
+    ///   - timer: seconds needed for the PIN visualization
+    public func setInput(input: NIInput, timer: Double, color: UIColor? = nil, completion: @escaping (NISuccessResponse?, NIErrorResponse?, @escaping () -> Void) -> Void) {
+        hideUI(true)
+        counter = timer
+        colorInput = color
+        viewModel = ViewPinViewModel(input: input)
+        viewModel?.callback = completion
+        activityIndicator.startAnimating()
+        GlobalConfig.shared.language = input.displayAttributes?.language
+        updateUI()
+    }
+    
+    // MARK: - Private
+
+    private func startTimer() {
+        timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(self.updateCounter), userInfo: nil, repeats: true)
+    }
+    
+    @objc private func updateCounter() {
+        if counter > 0 {
+            countDownDecription.text = "view_pin_countdown_description".localized + " " + String(Int(counter)) + " " + "view_pin_countdown_unit".localized
```

Comment:
Are you sure that string concatenation will be the same for RTL?

Reply:
<img width="337" alt="Screenshot 2023-11-01 at 14 29 11" src="https://github.com/network-international/card-management-sdk-ios/assets/72929424/2fb7ada6-de6f-4625-9ea5-e2770851032f">
<img width="338" alt="Screenshot 2023-11-01 at 14 29 24" src="https://github.com/network-international/card-management-sdk-ios/assets/72929424/a5d7df68-b7e6-4802-9c04-c16a608c74ef">

Did some tests and it looks good with phone in arabic (RTL)